rmw_fastrtps_shared_cpp/src/rmw_security.cpp

```diff
@@ -0,0 +1,368 @@
+  const std::string & file_name,
+  std::string & security_file_path)
+{
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  char * file_path = rcutils_join_path(node_secure_root.c_str(), file_name.c_str(), allocator);
+
+  if (!file_path) {
+    return false;
+  }
+
+  bool success = false;
+  if (rcutils_is_readable(file_path)) {
+    security_file_path = std::string(file_path);
+    success = true;
+  }
+
+  // This has been copied into security_file_path now-- safe to deallocate
+  allocator.deallocate(file_path, allocator.state);
+
+  return success;
+}
+
+std::string path_to_uri(const std::string & file_path)
+{
+  return std::string("file://") + file_path;
+}
+
+bool get_security_file_paths(
+  const std::string & node_secure_root, security_files_t & security_files)
+{
+  std::string file_path;
+
+  if (!get_security_file_path(node_secure_root, identity_ca_cert_file_name, file_path)) {
+    return false;
+  }
+  security_files.identity_ca_cert_path = file_path;
+
+  if (!get_security_file_path(node_secure_root, permissions_ca_cert_file_name, file_path)) {
+    return false;
+  }
+  security_files.permissions_ca_cert_path = file_path;
+
+  if (!get_security_file_path(node_secure_root, governance_file_name, file_path)) {
+    return false;
+  }
+  security_files.governance_path = file_path;
+
+  if (!get_security_file_path(node_secure_root, cert_file_name, file_path)) {
+    return false;
+  }
+  security_files.cert_path = file_path;
+
+  if (!get_security_file_path(node_secure_root, key_file_name, file_path)) {
+    return false;
+  }
+  security_files.key_path = file_path;
+
+  if (!get_security_file_path(node_secure_root, permissions_file_name, file_path)) {
+    return false;
+  }
+  security_files.permissions_path = file_path;
+
+  // Missing the logging configuration file is non-fatal
+  if (get_security_file_path(node_secure_root, logging_file_name, file_path)) {
+    security_files.logging_path = file_path;
+  }
+
+  return true;
+}
+}  // namespace
+
+#endif
+
+bool apply_logging_configuration_from_file(
+  const std::string & xml_file_path,
+  eprosima::fastrtps::rtps::PropertyPolicy & policy)
+{
+#if HAVE_SECURITY
+  tinyxml2::XMLDocument document;
+  document.LoadFile(xml_file_path.c_str());
+
+  auto log_element = document.FirstChildElement("security_log");
+  if (log_element == nullptr) {
+    RMW_SET_ERROR_MSG("logger xml file missing 'security_log'");
+    return RMW_RET_ERROR;
+  }
+
+  eprosima::fastrtps::rtps::PropertySeq properties;
+  add_property(
+    properties,
+    eprosima::fastrtps::rtps::Property(
+      logging_plugin_property_name,
+      "builtin.DDS_LogTopic"));
+
+  bool status = add_property_from_xml_element(
+    properties,
+    log_file_property_name,
+    *log_element,
+    "file");
+  if (!status) {
+    return status;
+  }
+
+  status = add_property_from_xml_element(
+    properties,
+    verbosity_property_name,
+    *log_element,
+    "verbosity");
```

Comment:
At the moment this does not append the `_LEVEL` suffix we discussed.

Reply:
Ah indeed, thanks for the reminder.

Reply:
Done.